pkg/state/state.go

```diff
@@ -2262,28 +2262,32 @@ func ConditionEnabled(r ReleaseSpec, values map[string]any) (bool, error) {
 	if len(r.Condition) == 0 {
 		return true, nil
 	}
-	conditionSplit := strings.Split(r.Condition, ".")
-	if len(conditionSplit) != 2 || conditionSplit[1] != "enabled" {
+	iValues := values
+	keys := strings.Split(r.Condition, ".")
+	if keys[len(keys)-1] != "enabled" {
 		return false, fmt.Errorf("Condition value must be in the form 'foo.enabled' where 'foo' can be modified as necessary")
 	}
-	if v, ok := values[conditionSplit[0]]; ok {
-		if v == nil {
-			return false, fmt.Errorf("environment values field '%s' is nil", conditionSplit[0])
-		}
-		vm, ok := v.(map[string]any)
+
+	currentKey := ""
+	for _, key := range keys[:len(keys)-1] {
+		currentKey = fmt.Sprintf("%s.%s", currentKey, key)
+		value, ok := iValues[key]
 		if !ok {
-			return false, fmt.Errorf("environment values field '%s' is not a map", conditionSplit[0])
+			return false, fmt.Errorf("environment values field '%s' not found", currentKey)
 		}
-		vv, ok := vm["enabled"]
+
+		iValues, ok = value.(map[string]interface{})
 		if !ok {
-			return false, nil
-		}
-		if vv == true {
-			return true, nil
+			return false, fmt.Errorf("environment values field '%s' is not a map", currentKey)
 		}
+	}
+
+	enabled, ok := iValues["enabled"].(bool)
+	if !ok {
 		return false, nil
 	}
-	return false, fmt.Errorf("environment values does not contain field '%s'", conditionSplit[0])
+
+	return enabled, nil
```

Comment:
please update the comments.

Reply:
What comments do you mean in this case? Are you talking about the comment block at the top of this function? Because I think with my changes those comments are still correct.

Reply:
oh. got it. @vandycknick